build/README.md

```diff
@@ -50,7 +50,7 @@ class CopyBuilder implements Builder {
   };
 
   @override
-  Future<void> build(BuildStep buildStep) async {
+  FutureOr<void> build(BuildStep buildStep) async {
```

Comment:
It's totally valid to change the return type to `Future<void>` since this is an async function – it awaits something.

Making a return type more specific in an override is fine.

Reply:
Yeah, you are absolutely right! It's fine to have a return type more specific in an override. But for a new user, it's confusing to have the implementation with `Future`

Reply:
> But for a new user, it's confusing to have the override with `Future`

It's always a bit of a struggle to choose between writing idiomatic code for the purposes of serving as a good example and something which can be copied, vs writing non-idiomatic code that elaborates things more than necessary.

In this case `Future<void>` was chosen over `FutureOr<void>` because it's more idiomatic to write it that way - as a definition of an signature that must be implemented `FutureOr` can be friendly, however as a definition of a concrete method signature it's preferable to tighten that to either `Future<void>` or `void` as appropriate. It is uncommon to have a synchronous builder so our examples focus on async builders, but they are possible so we do allow them.

I think for this example we should keep it as is since it will look more like the code that users will write.